src/server/scripts/Northrend/Naxxramas/boss_grobbulus.cpp

```diff
@@ -101,22 +123,65 @@ struct boss_grobbulus : public BossAI
     }
 };
 
-struct npc_grobbulus_poison_cloud : public ScriptedAI
+struct npc_grobbulus_poison_cloud : public NullCreatureAI
 {
-    npc_grobbulus_poison_cloud(Creature* creature) : ScriptedAI(creature)
+    npc_grobbulus_poison_cloud(Creature* creature) : NullCreatureAI(creature) { }
+
+    void JustAppeared() override
+    {
+        DoCastSelf(SPELL_POISON_CLOUD_PASSIVE);
+        DoCastSelf(SPELL_PACIFY_SELF);
+        me->DespawnOrUnsummon(61s);
+    }
+};
+
+struct npc_fallout_slime : public ScriptedAI
+{
+    npc_fallout_slime(Creature* creature) : ScriptedAI(creature)
     {
-        SetCombatMovement(false);
-        creature->SetReactState(REACT_PASSIVE);
+        Initialize();
     }
 
-    void IsSummonedBy(WorldObject* /*summoner*/) override
+    void Initialize()
     {
-        // no visual when casting in ctor or Reset()
-        DoCast(me, SPELL_POISON_CLOUD_PASSIVE, true);
-        DoCast(me, SPELL_PACIFY_SELF, true);
+        me->SetCorpseDelay(2, true);
+        me->SetReactState(REACT_PASSIVE);
+        DoCastSelf(SPELL_DISEASE_CLOUD);
+
+        _scheduler.Schedule(2s, [this](TaskContext /*task*/)
+        {
+            me->SetReactState(REACT_AGGRESSIVE);
+
+            if (Unit* summoner = ObjectAccessor::GetUnit(*me, _summonerGUID))
+                // maybe only if threat list was empty?
+                AddThreat(summoner, 100000.0f);
+        });
+    }
+
+    void IsSummonedBy(WorldObject* summoner) override
+    {
+        _summonerGUID = summoner->GetGUID();
+    }
+
+    void EnterEvadeMode(EvadeReason /*why*/) override
+    {
+        // they don't despawn when Grobbulus dies
```

Comment:
you can manually call Grobbulus summon hook to link each summon with the boss

something like `grobbulus->AI()->IsSummonedBy()`
I dont remember exactly which summon hook it is, maybe I'm wrong with this one

BossAI will take care to despawn and handle the summons

Reply:
`JustSummoned` this is the one

Reply:
They don't despawn when boss is killed because they are not related to boss in any way because they are summoned by player
Originally they just stuck after leaving combat(literally, they don't move), then they despawns after few seconds. I didn't recreated that behavior

Reply:
thats why I recommended the manual hook call, place a getcreature from instance to get grobbulus 👍

Reply:
`grobbulus->AI()->JustSummoned(slime)` will despawn them after boss death

Reply:
ah I finally realize what's the intended behaviour lul